### ISSUE TITLE
Add product-vs-personal-item CLS branch to student ReID model

### ISSUE THREAD

**Issue by joesu-angible** [joesu-angible/autoresearch#5]
## Summary
Add a product-vs-personal-item classification branch to the `student_finetune` ReID student model.

Target label contract:

```text
1 = 商品 / product
0 = 個人物品 / personal item / blacklist negative
```

This is inspired by the ReID+classification wiring in `angible/plasma`, but should be implemented as a small native branch in this repo rather than copied from plasma.

## Evidence from plasma

Checked local clone of `angible/plasma` at commit `f39fc30`.

Relevant pattern:

- `plasma/model.py` has standalone `EmbeddingModel` support for optional side logits:
  - `_prepare_embedding_head_cfg()` injects `productness_enabled=True` when `productness_bce_weight > 0`.
  - `_productness_targets_from_batch()` derives binary targets from `product_ids >= 0` and ORs in `is_commodity` when present.
  - `training_step()` / `validation_step()` pass `productness_logits` and `productness_targets` into the criterion only when supported.
- `plasma/config/emb/lcnet050_hier_pfc_supcon_f256_224.yaml` shows product/semantic/coarse supervision alongside embedding retrieval:
  - `model.name: EmbeddingModel`
  - `head.name: HierarchicalEmbeddingHead`
  - `loss.name: PartialFCSupConLoss`
  - product classification, semantic classification, product SupCon, pair SupCon, and teacher distillation weights.
- `plasma/export_wrappers.py` exports `(embeddings, productness_scores)` when the head emits `productness_logits`, by applying sigmoid.
- `tests/test_training_reporting.py` verifies validation logs include `val/loss/productness_bce` and `val/acc/productness_cls` when productness outputs are present.

Takeaway: plasma treats productness as an auxiliary side output over the same embedding backbone, not as a replacement for retrieval/distillation.

## Current autoresearch state

Checked local clone of `joesu-angible/autoresearch`, mainly:

- `student_finetune/train.py`
- `student_finetune/prepare.py`
- `student_finetune/train_final.py`
- `student_finetune/export_onnx.py`

Current facts:

- `LCNet` produces a 1280-d pooled summary and a normalized embedding via `self.proj`.
- Training already has multiple losses:
  - teacher distillation cosine loss;
  - optional ArcFace (`ArcMarginProduct`);
  - optional blacklist separation loss;
  - optional SSL / spatial / VAT losses.
- `CombinedDistillDataset` already mixes:
  - primary product roots: `REID_PRODUCTS`, `REID_COMMODITY`;
  - retail product root: `ARCFACE_DIR`;
  - blacklist root: `REID_NEGATIVES`.
- Current dataloader returns `(images, labels, paths)`, so there is not yet an explicit binary `is_product` target.
- `export_onnx.py` currently exports embedding-only output `[B, EMBEDDING_DIM]`.

### 1. Data contract

Extend distillation samples to carry productness labels:

```python
# product-like sources
REID_PRODUCTS   -> is_product = 1
REID_COMMODITY  -> is_product = 1
retail samples  -> is_product = 1

# negative / personal-item source
REID_NEGATIVES / blacklist -> is_product = 0
```

Recommended minimal API:

```python
CombinedDistillDataset.__getitem__ -> (image, label, path, is_product)
collate_distill -> (images, labels, paths, productness_targets)
```

Keep `labels` unchanged for existing ArcFace / separation behavior; add `productness_targets` as separate float tensor.

### 2. Model branch

Add a small head on top of the shared LCNet summary feature, before L2 normalization:

```python
self.productness_head = nn.Sequential(
    nn.Linear(1280, 256),
    nn.BatchNorm1d(256),
    nn.Hardswish(),
    nn.Dropout(p=0.1),
    nn.Linear(256, 1),
)
```

Expose training forward without breaking `encode()`:

```python
def forward_train(self, images):
    spatial, summary = self.forward_features(images)
    embedding = normalize(self.proj(summary))
    productness_logit = self.productness_head(summary).squeeze(1)
    return embedding, productness_logit
```

Keep `encode()` embedding-only for retrieval compatibility.

### 3. Loss

Use single-logit BCE, not 2-class CE:

```python
cls_loss = F.binary_cross_entropy_with_logits(productness_logits, productness_targets.float())
loss = distill_loss + arc_w * arc_loss + cls_w * cls_loss + ...
```

Suggested initial weight:

```python
PRODUCTNESS_CLS_WEIGHT = 0.02  # start conservative; tune 0.01-0.05
```

If negatives are rare, add `pos_weight` or balanced sampling later; do not start with a large weight that might collapse embeddings.

### 4. Metrics

Log separately from retrieval metrics:

- `productness_loss`
- `productness_acc`
- `productness_pos_acc` / product recall
- `productness_neg_acc` / personal-item rejection
- optionally AUC or FPR@TPR once validation split is formalized

Do not include cls accuracy in the existing `combined_metric` until calibrated, because current combined metric is retrieval+teacher alignment focused.

### 5. ONNX/export

Two modes are useful:

- default/backward-compatible: embedding-only output, unchanged;
- deployment mode: output `(embedding, productness_score)` where `score = sigmoid(logit)`.

This mirrors plasma's export wrapper behavior.

### 6. Validation

Need a separate productness validation split that includes both products and personal-item negatives. Retrieval validation should remain product-only or clearly scoped, otherwise personal-item negatives will pollute retrieval identity metrics.

### Phase A — minimal feature-flagged branch

- Add constants:
  - `USE_PRODUCTNESS_CLS = True`
  - `PRODUCTNESS_CLS_WEIGHT = 0.02`
  - `PRODUCTNESS_HEAD_HIDDEN = 256`
- Add `is_product` target plumbing through `CombinedDistillDataset` and `collate_distill`.
- Add `LCNet.productness_head` and a training forward path.
- Add BCE loss + logging in `run_train_epoch`.
- Include head params in optimizer and optimizer rebuild path.
- Keep `encode()` and retrieval eval unchanged.

### Phase B — validation and export

- Add productness eval over a held-out mix of product and personal-item crops.
- Add `--include-productness` to `export_onnx.py`.
- Verify ONNX numeric parity for both embedding and score output.

### Phase C — calibration/tuning

- Tune `PRODUCTNESS_CLS_WEIGHT` and sampling ratios.
- Track retrieval regression (`recall@1`, mean cosine) separately from productness gains.
- Consider hard-negative mining only after the baseline branch is stable.

## Risks / blockers

1. **Label definition risk**: `REID_NEGATIVES` must truly represent 個人物品/personal items. If it contains bad crops, background, or ambiguous product-like objects, the branch learns the wrong boundary.
2. **Sampling imbalance**: current `blacklist_ratio` may be low/zero, meaning the cls head may see too few negatives.
3. **Loss interference**: a large BCE weight can pull the shared backbone away from retrieval/distillation. Start low and monitor retrieval regression.
4. **Validation gap**: current retrieval eval does not measure personal-item rejection. Need a productness-specific eval before using it as a deployment gate.
5. **Export compatibility**: changing ONNX outputs by default can break existing consumers; make productness output opt-in first.

## Acceptance criteria

- Training still runs with `USE_PRODUCTNESS_CLS=False` exactly as before.
- With `USE_PRODUCTNESS_CLS=True`, dataloader returns productness labels and the model logs cls loss/accuracy.
- Retrieval metrics continue to run unchanged.
- Productness metrics are reported separately.
- ONNX embedding-only export remains backward compatible.
- Optional productness export outputs a score in `[0, 1]` and passes parity verification.

**Comment by joesu-angible** [comment]
Implemented and shipped via PR #8 (commit c2c4ac9). `ProductnessLCNet` adds a binary head on the shared 1280-d summary feature with `encode()` invariant preserved. ONNX export gains `--include-productness` (2-output: embedding + sigmoid score). Loss includes asymmetric label smoothing (eps_pos=0.05, eps_neg=0.02) + focal γ=2.0 to handle the noise pattern in REID_NEGATIVES that this issue flagged as the top risk. Mirrored on the teacher side too — DINOv3 trains a `DinoProductnessHead` so the embedding space is product-aware before distillation. GPU smoke confirmed end-to-end (`pos_acc=0.99`, `neg_acc=0.76` after 1 epoch, retrieval not regressed).
